packages/fdc3-get-agent/test/support/FrameTypes.ts

```diff
@@ -1,80 +1,94 @@
+  WebConnectionProtocol3Handshake,
+} from '@kite9/fdc3-schema/generated/api/BrowserTypes';
 
 /**
  * This handles the frame communications when we're using the embedded iframe approach
  */
-export function handleEmbeddedIframeComms(value: string, parent: MockWindow, cw: CustomWorld) {
-  const paramStr = value.substring(EMBED_URL.length + 1);
-  const params = new URLSearchParams(paramStr);
-  const connectionAttemptUuid = params.get('connectionAttemptUuid')!!;
-  const connection = cw.mockContext.getFirstInstance();
-  try {
-    parent.postMessage(
-      {
+export function handleEmbeddedIframeComms(_value: string, parent: MockWindow, source: MockWindow, cw: CustomWorld) {
+  const helloHandler = (e: Event) => {
+    const event = e as MessageEvent;
+    const eventSource = event.source as unknown as MockWindow;
+    const data = event.data;
+
+    if (isWebConnectionProtocol1Hello(data)) {
+      console.debug(
+        'Received hello message from: ',
+        eventSource.name,
+        eventSource == parent ? '(parent window): ' : '(NOT parent win): ',
+        event.data
+      );
+      const connection = cw.mockContext.getFirstInstance();
+
+      // send the other end of the channel to the app
+      const message: WebConnectionProtocol3Handshake = {
         type: 'WCP3Handshake',
         meta: {
-          connectionAttemptUuid: connectionAttemptUuid,
+          connectionAttemptUuid: data.meta.connectionAttemptUuid,
           timestamp: new Date(),
         },
         payload: {
           fdc3Version: '2.2',
-          intentResolverUrl: INTENT_RESOLVER_URL,
           channelSelectorUrl: CHANNEL_SELECTOR_URL,
+          intentResolverUrl: INTENT_RESOLVER_URL,
         },
-      } as WebConnectionProtocol3Handshake,
-      EMBED_URL,
-      [connection!!.externalPort]
-    );
-  } catch (e) {
-    console.error(e);
-  }
+      };
+      eventSource.postMessage(message, '*', [connection!.externalPort]);
+
+      window.removeEventListener('message', helloHandler);
+    } else {
+      console.warn(`Unexpected message received by MockIframe - ignored`, e);
+    }
+  };
+
+  //listen for hello message events from parent to respond to
+  source.addEventListener('message', helloHandler);
 }
 
 export function handleChannelSelectorComms(
   _value: string,
   parent: MockWindow,
-  source: Window,
+  source: MockWindow,
   cw: CustomWorld
 ): MessageChannel {
   const connection = new MessageChannel();
   try {
+    const msg: Fdc3UserInterfaceHello = {
+      type: 'Fdc3UserInterfaceHello',
+      payload: {
+        implementationDetails: 'mock channel selector',
+        initialCSS: {
+          width: '100px',
+        },
+      },
+    };
     parent.dispatchEvent({
       type: 'message',
-      data: {
-        type: FDC3_USER_INTERFACE_HELLO_TYPE,
-        payload: {
-          initialCSS: {
-            width: '100px',
-          },
-        },
-      } as Fdc3UserInterfaceHello,
+      data: msg,
```

Comment:
Is it possible to do something like this:

```ts
const event: Event = {
  type: "message",
  data: msg,
  origin: CHANNEL_SELECTOR_URL,
  source,
  ports: [connection.port1]
};
parent.dispatchEvent(event);
```

Just to avoid doing `as unknown as Event`

Reply:
This one is a pig. We can mock event, but that doesn't have the `data` property. The alternative is `MessageEvent`, which has a load of other properties we don't supply and doesn't like MockWindow as source.  Basically, anything else I've tried ends up needing at least as many overrides than this.

Best I've managed is:
```ts
const event: Event = new MessageEvent("message",
      {
        data: msg,
        origin: CHANNEL_SELECTOR_URL,
        source: source as unknown as MessageEventSource,
        ports: [connection.port1]
      }
    );
```

which isn't really any better.

Reply:
Oh well. We can leave it, then.